### PR TITLE
Chore: Fix legend changing when using incremental querying 

### PR DESCRIPTION
**Pull request by itsmylife**
**What is this feature?**

When we enable incremental querying and change the legend type it doesn't update the legend. 
So if we start querying the default is `auto`. If you change it to a custom legend `{{org}}` the legend doesn't reflect the change. That's because we were merging new and old frames behind the scenes. In that process, we used the old legends from old frames. This PR is making sure we update the legend information with the latest information. 

Also I renamed a bunch of variables. 
`targSigs` -> `targetSignatures`
`sig` -> `signature`
`ident` -> `identity`
That makes things quite easy to understand while reading the code. 

**Why do we need this feature?**

Consistent querying while incremental querying is enabled.

### CODE REVIEW

File: packages/grafana-prometheus/src/querycache/QueryCache.ts [hunk 1]
```diff
@@ -207,6 +207,7 @@ export class QueryCache<T extends SupportedQueryTypes> {
             if (amendedTable) {
               for (let i = 0; i < amendedTable.length; i++) {
                 cachedFrame.fields[i].values = amendedTable[i];
+                cachedFrame.fields[i].config = respFrame.fields[i].config;
```
Comment (leeoniya) [inline packages/grafana-prometheus/src/querycache/QueryCache.ts:210]: main thing to verify with this change is that `structureRev` does not increment excessively. it should stay stable across data refreshes when the field structures and config options do not change.

you can stick a `console.log()` into GraphNG to make sure we're not re-initializing the chart more than necessary.

https://github.com/grafana/grafana/blob/125a8a41dd6af4b5da86094d893d3bac429a6e6b/public/app/core/components/GraphNG/GraphNG.tsx#L210

Comment (itsmylife) [reply packages/grafana-prometheus/src/querycache/QueryCache.ts:216]: I did what you've suggested and it's just normal. Like it was before, so I'd say no impact on reinitialization.

**Review by leeoniya — DISMISSED** [review]

File: packages/grafana-prometheus/src/querycache/QueryCache.ts (hunk 1 above)
Comment (gtk-grafana) [reply packages/grafana-prometheus/src/querycache/QueryCache.ts:216]: I'm seeing a few issues here, first you should only update the displayNameFromDS. 
```
if(cachedFrame.fields[i].config.displayNameFromDS !== respFrame.fields[i].config.displayNameFromDS){
                  cachedFrame.fields[i].config.displayNameFromDS = respFrame.fields[i].config.displayNameFromDS;
                }
```

Initial query:
image

subsequent incremental queries:
image

Second, I'm seeing an issue with base incremental querying with rate queries.
I can't reproduce it in ops, but I can locally with the same datasource on this branch:
https://ops.grafana-ops.net/d/cdz3iu7slcvlsf/incremental-query-repro?from=now-12h&to=now&timezone=utc&refresh=&editPanel=1

Local:

Query:
```
sum(rate(adaptive_logs_bytes_dropped_total{job="loki-dev-005/distributor", container="distributor"}[$__rate_interval]))
```
DS: Dev-cortex

image
image

If we log out the field config, we can see that the cached field has an interval of 30s, and the respFrame has an interval of 10s, which is causing points to get unaligned, let me zoom in a bit on those screenshots:

image
image

Ok that doesn't really help does it, see the video below instead:


https://github.com/user-attachments/assets/b9890b50-e52d-4487-8da9-c7138c7ab45b

**Review by gtk-grafana — CHANGES_REQUESTED** [review]
Found a few bugs locally, don't merge quite yet!

**Comment by itsmylife** [comment]
@gtk-grafana I fixed the first issue. I cannot reproduce the second one. Let's take a look at it together when you are back.

**Comment by gtk-grafana** [comment]
@itsmylife I am seeing this behavior in ops today: https://ops.grafana-ops.net/d/cdz3iu7slcvlsf/incremental-query-repro?from=now-6h&to=now&timezone=utc. 

I think it's just a consequence of summing the rate over a smaller duration and concatenating that result with another sum of a rate, but not something introduced by this branch. We might want to create an issue for this, but this PR should be good to go now.

**Review by gtk-grafana — APPROVED** [review]
Looks good, works well locally. Thanks for the fix on this @itsmylife!

**Pull request merged**